client/api/client.go

```diff
@@ -0,0 +1,110 @@
+package api
```

Comment:
Not a fan of the package name "api".
We should name it "gateway", which will be focused on interacting with gateway api as client.